### ISSUE TITLE
Ability to Setup Custom HitBoxes For Entities

### ISSUE THREAD

**Issue by EthanTheGreat** [Facepunch/garrysmod-requests#620]
**Description**
Right now, there is absolutely no way to set custom hitboxes. The only hitboxes available are the model's hitbox structures. I want the ability to create custom hitboxes, on the spot.

**Example**
`SetupHitBoxStructure` or `SetupHitBoxes`
- The idea behind the concept would allow the ability to edit an entity and its hitboxes.
- Ray-Traces wouldn't be dependent on an entities physical mesh.
- Hitboxes could be made up for custom entities.
- Custom Hitboxes would be separate from the model's precached data.

**Why?**
The following function is our `only` way to set hitboxes.
http://wiki.garrysmod.com/page/Entity/SetHitboxSet

There isn't any documentation, but from what I've gathered, the function doesn't seem to work as I wanted. It uses the precached data and seems very impractical for use.

Currently, there isn't any existing way to setup custom hitboxes without recompiling the model.

Please consider including this in the next update.
Thank you.

**Comment by meepdarknessmeep** [comment]
i believe this is all controlled by files and can not really be set in code

**Comment by EthanTheGreat** [comment]
Yup, I can't seem to think of alternative way.. Aside from just recompiling the model itself.

**Comment by Askastaa** [comment]
and where do i find half life models such as manhack
